scripts/pretraining/bert/run_pretraining.py

```diff
@@ -0,0 +1,394 @@
+    cfg, tokenizer, model = get_pretraining_model(args.model_name, ctx_l, args.max_seq_length)
+
+    if args.raw:
+        get_dataset_fn = functools.partial(get_pretrain_data_text,
+                                           max_seq_length=args.max_seq_length,
+                                           short_seq_prob=args.short_seq_prob,
+                                           masked_lm_prob=args.masked_lm_prob,
+                                           max_predictions_per_seq=args.max_predictions_per_seq,
+                                           whole_word_mask=args.whole_word_mask,
+                                           tokenizer=tokenizer,
+                                           circle_length=args.circle_length,
+                                           repeat=args.repeat,
+                                           dataset_cached=args.dataset_cached,
+                                           num_max_dataset_cached=args.num_max_dataset_cached)
+    else:
+        get_dataset_fn = get_pretrain_data_npz
+
+    data_train = get_dataset_fn(args.data, args.batch_size, shuffle=True,
+                                num_buckets=args.num_buckets, vocab=tokenizer.vocab,
+                                num_parts=num_workers, part_idx=rank,
+                                num_dataset_workers=args.num_dataset_workers,
+                                num_batch_workers=args.num_batch_workers)
+
+    param_dict = model.collect_params()
+    # Do not apply weight decay to all the LayerNorm and bias
+    for _, v in model.collect_params('.*beta|.*gamma|.*bias').items():
+        v.wd_mult = 0.0
+    # Set grad_req if gradient accumulation is required
+    params = [p for p in param_dict.values() if p.grad_req != 'null']
+    num_accumulated = args.num_accumulated
+    if num_accumulated > 1:
+        logging.info('Using gradient accumulation. Effective global batch size = {}'
+                     .format(num_accumulated * args.batch_size * len(ctx_l) * num_workers))
+        for p in params:
+            p.grad_req = 'add'
+
+    num_steps = args.num_steps
+    warmup_steps = int(num_steps * args.warmup_ratio)
+    log_interval = args.log_interval
+    save_interval = args.ckpt_interval
+    logging.info('#Total Training Steps={}, Warmup Steps={}, Save Interval={}'
+                 .format(num_steps, warmup_steps, save_interval))
+    lr_scheduler = PolyScheduler(max_update=num_steps,
+                                 base_lr=args.lr,
+                                 warmup_begin_lr=0,
+                                 pwr=1,
+                                 final_lr=0,
+                                 warmup_steps=warmup_steps,
+                                 warmup_mode='linear')
+    optimizer_params = {'learning_rate': args.lr,
+                        'wd': args.wd,
+                        'lr_scheduler': lr_scheduler,
+                        }
+    if args.optimizer == 'adamw':
```

Comment:
We can choose to use lamb by default: https://mxnet.apache.org/versions/1.6/api/python/docs/api/optimizer/index.html#mxnet.optimizer.LAMB

Reply:
lamb suites for large batch size.